pkg/monitor/cluster/cluster.go

```diff
@@ -212,35 +211,50 @@ func (mon *Monitor) Monitor(ctx context.Context) (errs []error) {
 		})
 	}
 
-	//this API server healthz check must be first, our geneva monitor relies on this metric to always be emitted.
-	statusCode, err := mon.emitAPIServerHealthzCode(ctx)
+	timeCall := func(ctx context.Context, f func(context.Context) error) error {
+		innerNow := time.Now()
+		collectorName := steps.ShortName(f)
+		mon.log.Debugf("running %s", collectorName)
+		innerErr := f(ctx)
+		if innerErr != nil {
+			e := fmt.Errorf("failure running cluster collector '%s': %w", collectorName, innerErr)
+			mon.log.Debug(e.Error())
+			// emit metrics collection failures and collect the err, but
+			// don't stop running other metric collections
+			mon.emitMonitorCollectorError(collectorName)
+			return e
+		} else {
+			timeToComplete := time.Since(innerNow).Seconds()
+			mon.emitMonitorCollectionTiming(collectorName, timeToComplete)
+			mon.log.Debugf("successfully ran cluster collector '%s' in %2f sec", collectorName, timeToComplete)
+		}
+		return innerErr
+	}
```

Comment:
```suggestion
func (mon *Monitor) timeCall(ctx context.Context, collector func(context.Context) error) error {
	startTime := time.Now()
	collectorName := steps.ShortName(collector)

	mon.log.Debugf("running %s", collectorName)
	err := collector(ctx)

	if err != nil {
		wrappedErr := fmt.Errorf("failure running cluster collector '%s': %w", collectorName, err)
		mon.log.Debug(wrappedErr.Error())
		mon.emitMonitorCollectorError(collectorName)
		return wrappedErr
	}

	duration := time.Since(startTime).Seconds()
	mon.emitMonitorCollectionTiming(collectorName, duration)
	mon.log.Debugf("successfully ran cluster collector '%s' in %.2f sec", collectorName, duration)

	return nil
}
```